### PR TITLE
EPMRPP-84522 || Added activity API specification

### PR DESCRIPTION
**Pull request by rkukharenka**

### CODE REVIEW

**Comment by APiankouski** [comment]
Imported this spec to https://reportportal.stoplight.io/studio/epm-rpp?source=7yVd9arsLBFzOYFUENDZQ

File: api/openapi/activities.yaml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            id: kr5ju3d2ftb61
+        old_value:
+          type: string
+          x-stoplight:
+            id: 9zzt3diczvwnc
+        new_value:
+          type: string
+          x-stoplight:
+            id: gcydbgmi7or7q
+    SearchCriteria:
```
Comment (raikbitters) [inline api/openapi/activities.yaml:166]: @rkukharenka shall we divide it into a sharing model?

**Review by AmsterGet — COMMENTED** [review]
Following the [guideline](https://opensource.zalando.com/restful-api-guidelines/#get-with-body) you've provided I assume that the `POST with body payload` option can be considered **only** when a `GET with URL encoded query parameters` is not possible.
For now I don't see such limitations and we didn't receive such cases from our users.

image

Additionally if we consider such filtering interface as a standard, we need to plan the same migration for some other endpoints (f.e. `/v1/{projectName}/item`,  `/v1/{projectName}/item/history`) where a lot of query parameters possible.

cc @Bam6ycha

**Review by APiankouski — APPROVED** [review]

File: api/openapi/activities.yaml (hunk 1 above)
Comment (rkukharenka) [reply api/openapi/activities.yaml:166]: I think we can move this to a sharing model if the need arises in the future

**Comment by rkukharenka** [comment]
> 

Filtering has complex query criteria and for large queries it is recommended to use the RequestBody. Sending a request with more than 10 request parameters looks overloaded. This can also be viewed in REST API Design from [Atatus](https://www.atatus.com/blog/rest-api-design-filtering-sorting-and-pagination/#:~:text=on%20complex%20conditions.-,3.%20Request%20Body,-Occasionally%2C%20you%20may).

Regarding such endpoints as /v1/{projectName}/item, /v1/{projectName}/item/history it would also be nice to rework in the future.

cc: @raikbitters , @AmsterGet

**Pull request merged**